### PR TITLE
DCOS-28689: cli and programmatic usage

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
Separate CLI and Programmatic usage

Would it make sense to enhance the docs?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,16 +92,27 @@
...
   },
   "commitlint": {
-    "extends": ["@commitlint/config-conventional"],
+    "extends": [
+      "@commitlint/config-conventional"
+    ],
     "rules": {
-      "scope-case": [0]
+      "scope-case": [
+        0
```
Comment (juliangieseke) [inline package.json]: 🤔 ?

File: src/index.ts [hunk 2]
```diff
@@ -1,20 +1,5 @@
...
-
-let config;
-
-try {
-  config = getConfig();
-} catch (e) {
-  log.error("setup", e.message);
-  process.exit(1);
-}
+import config from "./config";
```
Comment (juliangieseke) [inline src/index.ts]: this shouldnt be included in the programmatic part 🤔 

the current implementation of `config` is "get stuff from the environment" - which is only feasable in cli usage. instead we should implement some kind of 'programmatic config gatherer' or just export the `IServerConfig` Interface & `server` method.

actually I'm not even sure if this file is needed at all… instead we could just use `server.ts` directly for programatic usage (like we do in `cli.ts` 🤔 )

File: package.json (hunk 1 above)
Comment (DanielMSchmidt) [reply package.json]: img

**Comment by nLight** [comment]
Flicts!

**Comment by DanielMSchmidt** [comment]
@nLight solved the conflict 👍

**Comment by DanielMSchmidt** [comment]
@juliangieseke could you re-review this?

File: __tests__/helpers/mockServerController.ts [hunk 3]
```diff
@@ -1,53 +1,33 @@
-const { spawn } = require("child_process");
+import mockserver, { IServerHandle } from "../../src/server";
```
Comment (nLight) [inline __tests__/helpers/mockServerController.ts]: with typescript we can setup local module resolution and stop doing relative paths. What do you think?

Comment (DanielMSchmidt) [reply __tests__/helpers/mockServerController.ts]: I tried it, but it is not possible.

Comment (nLight) [reply __tests__/helpers/mockServerController.ts]: Tell me more :)

Comment (DanielMSchmidt) [reply __tests__/helpers/mockServerController.ts]: You can set different roots with typescript, but that just works with modules, so I would need to write `from "server"` instead, which is way too much magic IMHO. Here is the handbook article I read for that: http://www.typescriptlang.org/docs/handbook/module-resolution.html

Comment (DanielMSchmidt) [reply __tests__/helpers/mockServerController.ts]: @nLight is this enough info for you or would you like to get it working together today?

**Comment by nLight** [comment]
> Would it make sense to enhance the docs?
yes, please

**Comment by DanielMSchmidt** [comment]
@nLight wrote some documentation, could you take a look?

**Pull request merged**